spec/service_consumers/pact_helper.rb

```diff
@@ -106,6 +106,53 @@ def url_encode(str)
     end
   end
 
+  provider_state "there is a valid user session, with saved pages" do
+    set_up do
+      SavedPage.destroy_all
+      user = OidcUser.find_or_create_by(sub: "user-id")
```

Comment:
this feels particularly wrong but I'm getting:
```
     ActiveRecord::RecordNotUnique:
       PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "index_oidc_users_on_sub"
       DETAIL:  Key (sub)=(user-id) already exists.
```

if i switch it to:
```
      user = FactoryBot.create(:oidc_user, sub: "user-id")
```

Reply:
That's because `FactoryBot.create(:oidc_user, sub: "user-id")` is trying to recreate the user record which may already exist.

Reply:
Why is this `SavedPage.destroy_all` needed?

Reply:
But shouldn't the test database be empty at this point?
I don't see why that record should already exist?

Reply:
I don't think it runs each test in a transaction

Reply:
what do you think the best approach is? all these setup and teardown blocks seem a bit lame?

I could create a user once at the top perhaps?

Reply:
> I could create a user once at the top perhaps?

Yes, my fixup commit does that, and makes all the tests use it instead of specifying `user-id` everywhere